src/functions.jl

```diff
@@ -114,8 +114,14 @@ for (mod, funs) in (
                 return NamedDimsArray{L}(data)
             end
 
+            function $mod.$fun(a::AbstractArray, b::NamedDimsArray{L}) where L
+                data = $mod.$fun(a, parent(b))
+                newL = L[1:ndims(a)]
+                return NamedDimsArray{newL}(data)
+            end
+
             function $mod.$fun(a::NamedDimsArray{La}, b::NamedDimsArray{Lb}) where {La, Lb}
-                newL = unify_names_longest(La, Lb)
+                newL = unify_names_longest(La, Lb)[1:ndims(a)]
```

Comment:
Does that optimize away?

Reply:
Maybe not, sadly?
```
m = rand(2,2);
@btime (m -> (1,2,3,4)[1:ndims(m)])($m) # 274.245 ns (2 allocations: 128 bytes)
```

Reply:
Maybe I give up and just return `a`, without filling in any missing names it may have.

Reply:
Maybe just adapt the `unify_named_longest` 
to be either `unify_named_shortest` (will it error if the first one isn't shortest anyway?)
or `unify_names_first`

That code is in `name_core.jl`

Reply:
Good idea, done. 

Notice that this means we don’t exactly return the mutated `a`. But still another wrapper of the same data.

Reply:
> Notice that this means we don’t exactly return the mutated a. But still another wrapper of the same data.

That is inline I feel with how it should be,
getting rid of wild-card names where possible